libraries/AP_UAVCAN/AP_UAVCAN.cpp

```diff
@@ -567,6 +847,37 @@ void AP_UAVCAN::do_cyclic(void)
 
                 rc_out_sem_give();
             }
+
+            uint64_t c_send_time = AP_HAL::millis64();
+
+            if (fix_out_sem_take()) {
+                if (fix_out_array[_uavcan_i] != nullptr && _broadcast_fix_rate) {
```

Comment:
(_broadcast_fix_rate > 0) is better. Saves you from dealing with the abs later

Reply:
Even better, make it all one if:
```
if (fix_out_array[_uavcan_i] != nullptr &&
  _broadcast_fix_rate > 0 &&
 (now - fix_out_last_send_time) > (1000 / _broadcast_fix_rate)) {
```

.... and even BETTER is to avoid the expensive 1kHz division calc in the Hz to ms conversion on each of these and store it in a different variable
```
if (fix_out_array[_uavcan_i] != nullptr &&
_broadcast_fix_rate > 0 &&
now >= fix_out_next_send_time_ms) {
    fix_out_next_send_time_ms = now + (1000 / _broadcast_fix_rate);
```